pokemon-spring-mvc/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/mvc/controllers/AdminTrainerController.java

```diff
@@ -0,0 +1,134 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.mvc.controllers;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.TrainerCreateDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.NoAdministratorException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.facade.TrainerFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.util.NestedServletException;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.inject.Inject;
+import javax.validation.ConstraintDeclarationException;
+import javax.validation.ConstraintViolationException;
+import javax.validation.Valid;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ResourceBundle;
+
+/**
+ * Controller class for Trainer for Admins.
+ *
+ * @author Michal Mokros 456442
+ */
+@Controller
+@RequestMapping("/admin/trainer")
+public class AdminTrainerController {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(AdminTrainerController.class);
+    ResourceBundle messages = ResourceBundle.getBundle("Texts", LocaleContextHolder.getLocale());
+
+    @Inject
+    TrainerFacade trainerFacade;
+
+    /**
+     * Get Controller for creating new Trainer
+     * @param model data to be displayed
+     * @return JSP page
+     */
+    @RequestMapping(value = "/new", method = RequestMethod.GET)
```

Comment:
Note: mapping get and post to different urls (new vs. create) is tricky. It actually wasn't done correctly in PokemonSpecies and the easiest fix was to use just 1 url, so if you based this on PokemonSpecies then I'd suggest remaking this to use just the 1 url.